frontend/src/components/SideNavBar.tsx

```diff
@@ -1,15 +1,19 @@
+"use client";
+import { useState } from 'react';
 import React from 'react'
 import svgIcons from "../app/svgIcons";
 import SideNavBarButton from "@/components/SideNavBarButton";
 export default function SideNavBar() {
+  const [currentSelection, setCurrentSelection] = useState("Home");
+  const updatedText = currentSelection;
```

Comment:
Is this needed? currentSelection has the state, and it will change as they click the buttons. I don't think having an extra variable here matters.

Reply:
In fact this may be why useEffect is there too... just use the currentSelection, and the props will update as needed.

Reply:
just saw this second comment. I'll get that fixed.